smartmin/views.py

```diff
@@ -27,6 +29,24 @@
 from smartmin.csv_imports.models import ImportTask
 from . import widgets
 
+
+def build_email_context(request=None, user=None):
+    context = Context({'user': user})
+
+    processors = []
+    collect = []
+    collect.extend(('smartmin.users.context_processors.link_components',))
+    collect.extend(getattr(settings, "EMAIL_CONTEXT_PROCESSORS", ()))
+    for path in collect:
+        func = import_string(path)
+        processors.append(func)
+
+    for processor in processors:
+        context.update(processor(request))
+
+    return context
+
+
```

Comment:
And the extend behavior is still goofy here.

Basically we want it to work like this:

- If they don't have EMAIL_CONTEXT_PROCESSORS set, then that defaults to smartmin.email.link_components
- If they do, then we use whatever is in there and do not prepend or append anything at all

The processors should also be in /smartmin/email